src/Parser/AnonymousClassVisitor.php

```diff
@@ -27,10 +27,20 @@ public function enterNode(Node $node): ?Node
 			return null;
 		}
 
-		$node->setAttribute(self::ATTRIBUTE_ANONYMOUS_CLASS, true);
+		$node = new AnonymousClassNode(
+			$node->name,
+			[
+				'flags' => $node->flags,
+				'extends' => $node->extends,
+				'implements' => $node->implements,
+				'stmts' => $node->stmts,
+				'attrGroups' => $node->attrGroups,
+			],
```

Comment:
This feels a bit brittle.. if this array is ever extended we'll have a "broken" class node. This might e.g. happen if PHP adds support for generics.

Maybe this could serve as a case to demonstrate to https://github.com/nikic/PHP-Parser that an alternative implementation with e.g. a separate flag might be more optimal.
It's a bit unclear what the backward compatibility promise is there, so I'll just ask.